lib/command.ts

```diff
@@ -156,6 +156,7 @@ export default class Command implements ICommand {
   public isCustomCommand = false;
   public inTransaction = false;
   public pipelineIndex?: number;
+  private _commandTimeoutTimer?: ReturnType<typeof setTimeout>;
```

Comment:
Should be able to just use `NodeJS.Timeout` as the type

Reply:
> Looks ok, but I'm not sure if it's necessary.
> Are the timers causing actual measurable performance problems or other issues without this change?
> 
> If we start clearing them just in case, we should also consider the cost of increased code complexity (maintenance overhead, wider surface area for possible future bugs etc).
> Sometimes an optimization saves a few microseconds of CPU time, but can cost hours of developer time at some point in the future.
> Might not be the case here, but just something to consider 🙂

If you have a lingering timeout per command and you're doing several commands I'd say it's relevant and the additional complexity for this single operation is marginal, but it's up to you.

Regarding the type change, I can do as your suggested if you feel it's a better fit than using return type. I had tried Timeout without the namespace and number as well and ended up here.

Reply:
updated the type as suggested.